h2o-bindings/bin/custom/python/gen_infogram.py

```diff
@@ -0,0 +1,230 @@
+            return None
+        else:
+            return self._model_json["output"]["all_predictor_names"], self._model_json["output"]["relevance"]
+
+    def get_all_predictor_cmi(self):
+        """
+        Get normalized cmi of all predictors.
+        :param self: 
+        :return: two tuples, first one is predictor names and second one is cmi
+        """
+        if self._model_json["output"]["all_predictor_names"] is None:
+            return None
+        else:
+            return self._model_json["output"]["all_predictor_names"], self._model_json["output"]["cmi"]
+
+    def get_all_predictor_cmi_raw(self):
+        """
+        Get raw cmi of all predictors.
+        :param self: 
+        :return: two tuples, first one is predictor names and second one is cmi
+        """
+        if self._model_json["output"]["all_predictor_names"] is None:
+            return None
+        else:
+            return self._model_json["output"]["all_predictor_names"], self._model_json["output"]["cmi_raw"]
+        
+    # Override train method to support infogram needs
+    def train(self, x=None, y=None, training_frame=None, verbose=False, **kwargs):
+        sup = super(self.__class__, self)
+        
+        def extend_parms(parms):  # add parameter checks specific to infogram
+            if parms["data_fraction"] is not None:
+                assert_is_type(parms["data_fraction"], numeric)
+                assert parms["data_fraction"] > 0 and parms["data_fraction"] <= 1, "data_fraction should exceed 0" \
+                                                                                   " and <= 1."
+        
+        parms = sup._make_parms(x,y,training_frame, extend_parms_fn = extend_parms, **kwargs)
+ 
+        sup._train(parms, verbose=verbose)
+        # can probably get rid of model attributes that Erin does not want here
+        return self
+
+extensions = dict(
+    __imports__="""
+import ast
+import json
+import warnings
+import h2o
+from h2o.utils.typechecks import assert_is_type, is_type, numeric
+from h2o.frame import H2OFrame
+import numpy as np
+""",
+    __class__=class_extensions
+)
+       
+overrides = dict(
+    algorithm_params=dict(
+        getter="""
+if self._parms.get("{sname}") != None:
+    algorithm_params_dict =  ast.literal_eval(self._parms.get("{sname}"))
+    for k in algorithm_params_dict:
+        if len(algorithm_params_dict[k]) == 1: #single parameter
+            algorithm_params_dict[k] = algorithm_params_dict[k][0]
+    return algorithm_params_dict
+else:
+    return self._parms.get("{sname}")
+""",
+        setter="""
+assert_is_type({pname}, None, {ptype})
+if {pname} is not None and {pname} != "":
+    for k in {pname}:
+        if ("[" and "]") not in str(algorithm_params[k]):
+            algorithm_params[k] = [algorithm_params[k]]
+    self._parms["{sname}"] = str(json.dumps({pname}))
+else:
+    self._parms["{sname}"] = None
+"""
+    ),
```

Comment:
I think I suggested to remove this, but previous comment was probably lost after the last renaming.
If it doesn't work after deleting this, then the proper way to handle this param is explained here: https://github.com/h2oai/h2o-3/pull/5572/files#r765268303

No need for back and forth translation, simpler and safer.

Reply:
@sebhrusen I tried your method and it does not work.  Since I am going to re-implement this part in JIRA:  https://h2oai.atlassian.net/browse/PUBDEV-8465, I would like to fix it there.  The problem is we treat algorithm_params in Java as string.  Hence, in part of python it is a string,  The setter/getter allows people to enter it as python dict and translate back and forth between the two.  The problem is before I call the infogram.train method, when I am just initializing the thing here:  

    gbm_params = {'ntrees':3, 'max_depth':5}
    infogram_model_gbm = H2OInfogram(seed = 12345, top_n_features=50,
                                     algorithm='gbm', algorithm_params=gbm_params)

Now the algorithm_params is a dict and not string.  However the getter/setter combo takes are of this case and make it work.  Anyway, with the new implementation in the JIRA, these will all go away.